snuba/snuba_migrations/transactions/0017_transactions_add_app_start_type_column.py

```diff
@@ -0,0 +1,51 @@
+from typing import Sequence
+
+from snuba.clickhouse.columns import Column, String
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations
+from snuba.migrations.columns import MigrationModifiers as Modifiers
+
+
+class Migration(migration.ClickhouseNodeMigration):
+    """
+    The app start type column is required to query for different
+    app start types.
+    """
+
+    blocking = False
+
+    def __forward_migrations(
+        self, table_name: str
+    ) -> Sequence[operations.SqlOperation]:
+        return [
+            operations.AddColumn(
+                storage_set=StorageSetKey.TRANSACTIONS,
+                table_name=table_name,
+                column=Column(
+                    "app_start_type",
+                    String(Modifiers(low_cardinality=True)),
```

Comment:
Is app_start_type field user generated or machine generated? If it is machine generated and would have mostly a few values then having it as low_cardinality is ok. If it is user generated, what is the likelihood of having high cardinality of this value?

Reply:
The SDKs create these values, and it's not exposed to the user. Currently, we have four different values. The likelihood of this having a high cardinality is very low. Maybe it increases to 10 or 20 at some point, but not more than 100 for sure.